### PR TITLE
Process data model

### PR DESCRIPTION
**Pull request by fernando-silvabr66**
- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.
- Create custom methods, for:
> - Users - A method that returns the 3 most recent posts for a given user.
> - Posts - A method that updates the posts counter for a user and a method that returns the 5 most recent comments for a given post.
> - Comments - A method that updates the comments counter for a post.
> - Likes - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/post.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again the vice versa. Simply do:
```ruby
# app/models/post.rb
belongs_to :author, class_name: 'User'

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/like.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again the vice versa. Simply do:
```ruby
# app/models/like.rb
belongs_to :author, class_name: 'User'

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] Foreign key already declared in the _User_ model, and hence no need to declare it again the vice versa. Simply do:
```ruby
# app/models/comment.rb
belongs_to :author, class_name: 'User'

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Fernando, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job so far implementing association among models


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**

- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by sole contributor. 
Implementing the above change(s) will make your ReadMe appear more professional.


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (fernando-silvabr66) [reply app/models/post.rb]: Fixed.

File: app/models/like.rb (hunk 2 above)
Comment (fernando-silvabr66) [reply app/models/like.rb]: Fixed.

File: app/models/comment.rb (hunk 3 above)
Comment (fernando-silvabr66) [reply app/models/comment.rb]: Fixed.

**Comment by fernando-silvabr66** [comment]
Thanks for the suggestions.

**Review by BenMukebo — APPROVED** [review]
### Hi @fernando-silvabr66, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- The associations between models is set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).:heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL]  Please, add a meaningful  `Project description`. Remember A project description is a high-level overview of why you're doing a project. The document explains a project's objectives and essential qualities. Is an essential document that provides your audience with a comprehensive overview of the project. Think of it as the elevator pitch that focuses on what and why without delving into how. 😉 💯 

- The README file It is the first file a person will see when they encounter your project, so it should be fairly brief but detailed. It will make your project stand out from a bunch of others.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**